lib/src/jpegr.cpp

```diff
@@ -628,6 +693,67 @@ uhdr_error_info_t JpegR::generateGainMap(uhdr_raw_image_t* sdr_intent, uhdr_raw_
   generateMap();
   std::for_each(workers.begin(), workers.end(), [](std::thread& t) { t.join(); });
 
+  /*refine_min_max_content_boost(gainmap_data, gainmap_min, gainmap_max, map_width, map_height,
+                               map_width, mUseMultiChannelGainMap);*/
+
+  float min_content_boost_log2 = gainmap_min[0];
+  float max_content_boost_log2 = gainmap_max[0];
+  for (int index = 1; index < (mUseMultiChannelGainMap ? 3 : 1); index++) {
+    min_content_boost_log2 = (std::min)(gainmap_min[index], min_content_boost_log2);
+    max_content_boost_log2 = (std::max)(gainmap_max[index], max_content_boost_log2);
+  }
+
+  std::function<void()> encodeMap = [this, gainmap_data, map_width, dest, min_content_boost_log2,
```

Comment:
This change has added an additional step and I'm afraid that this may affect the performance. Could you try some profiling for generateGainMap() call before and after this change?

(We were doing this before but changed it to one step as the current version at some point, reasons are performance consideration, and below)

Also note that we saw this may NOT ideal for image quality especially when there is JPEG input because the JPEG input may have encoding artifacts